pkg/utils/validation/dataset.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2024 The Fluid Author.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"strings"
+
+	"github.com/fluid-cloudnative/fluid/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func IsValidDataset(dataset v1alpha1.Dataset, enableMountValidation bool) error {
+	if errs := validation.IsDNS1035Label(dataset.ObjectMeta.Name); len(dataset.ObjectMeta.Name) > 0 && len(errs) > 0 {
+		return field.Invalid(field.NewPath("metadata").Child("name"), dataset.ObjectMeta.Name, strings.Join(errs, ","))
+	}
+
+	// 0.1 Validate the mount name and mount path
+	// Users can set the environment variable to 'false' to disable this validation
+	// Default is true
+	if !enableMountValidation {
```

Comment:
Do we have to add a validation option here? WDYT @cheyang @zhang-x-z

Reply:
I think it's not essential. Instead, how about putting the validation logic into function ReconcileInternal？

https://github.com/fluid-cloudnative/fluid/blob/2723eedbc5542a811dce3f24fba18eedb2c55f84/pkg/controllers/runtime_controller.go#L83